mongomock/aggregate.py

```diff
@@ -745,8 +790,18 @@ def _handle_count_stage(in_collection, database, options):
     return [{options: len(in_collection)}]
 
 
+def _handle_add_fields_stage(in_collection, unused_database, options):
+    for doc in in_collection:
+        for key, value in options.items():
+            try:
+                doc[key] = _parse_expression(value, doc)
+            except KeyError:
+                pass
+    return in_collection
+
+
 _PIPELINE_HANDLERS = {
-    '$addFields': None,
+    '$addFields': _handle_add_fields_stage,
```

Comment:
Please move that to a different MR: you already have a lot in here.

Reply:
We really need the `$addFields` operator... can you please push it separately?

Reply:
also, it seems like the $addFields solution doesn't support nested params, such as:
```

Reply:
The $addFields has been added in 9b92a7e018